tests/Rector/ClassMethod/AddRouteAnnotationRector/Fixture/fixture.php.inc

```diff
@@ -26,7 +26,7 @@ use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 class AppController extends Controller
 {
     /**
-     * @\Symfony\Component\Routing\Annotation\Route(path="/all/{foo}/{page}", name="all", defaults={foo="foo123", page=1}, host="m.example.com", schemes={"http", "https"}, methods={"GET", "POST"}, condition="context.getMethod() in ['GET', 'HEAD'] and request.headers.get('User-Agent') matches '/firefox/i'", requirements={page="\d+"})
+     * @\Symfony\Component\Routing\Annotation\Route(path="/all/{foo}/{page}", name="all", defaults={"foo"="foo123", "page"=1}, host="m.example.com", schemes={"http", "https"}, methods={"GET", "POST"}, condition="context.getMethod() in ['GET', 'HEAD'] and request.headers.get('User-Agent') matches '/firefox/i'", requirements={"page"="\d+"})
```

Comment:
You're welcome :)

Yes, good idea

Reply:
I did a little research in my prototype project. As you see in https://github.com/malteschlueter/symfony-reproducers/blob/058d6ad3d4567bf1caf5754ec030b974cfe371e2/config/routes.yaml#L11-L12 I've added two integers keys to the route configuration. When I run `php bin/console debug:router --format=json` and look at the compiled configuration the two integer keys have been changed.
```json
    "foo_preview_parameter": {
        "path": "\/foo\/preview\/parameter\/{foo}\/{page}",
        "pathRegex": "#^\/foo\/preview\/parameter(?:\/(?P<foo>[^\/]++)(?:\/(?P<page>\\d+))?)?$#sD",
        "host": "ANY",
        "hostRegex": "",
        "scheme": "http|https",
        "method": "ANY",
        "class": "Symfony\\Component\\Routing\\Route",
        "defaults": {
            "_controller": "FooPreviewBundle:Preview:parameter",
            "foo": "test123",
            "page": 1,
            "0": 23,
            "1": 23
        },
        "requirements": {
            "page": "\\d+"
        },
        "options": {
            "compiler_class": "Symfony\\Component\\Routing\\RouteCompiler"
        }
    }
```

I figured out that a `array_merge` in the `RouteCollectionBuilder` cause this https://github.com/symfony/symfony/blob/0a9296a8b23360475c282cd98b1fd36e3937d259/src/Symfony/Component/Routing/RouteCollectionBuilder.php#L273.

---

So I'm not sure what to do :/ I would say we ignore integer keys because it's very unusal to set for routes. Isn't it?

Reply:
I see. I meant adding test case for the quoting service part, unrelated to routing annotation per-se.

In general, the numbers should be skipped from quoting. I see it in the code, just not in the test cases. In future, we might remove the string check and tests could pass false positively. So just to avoid it :)

Reply:
Do you mean to add a test case for https://github.com/rectorphp/rector-symfony/blob/88dd052834e54417ac5222171b47f553b5365f3c/src/Rector/ClassMethod/AddRouteAnnotationRector.php#L129-L141? Like extract it to a own service to make it testable?

Reply:
The `StringValueQuoteWrapper` was good hint but I was a little bit scared to refactored it and destroy something. Because of that I created a new service and used it for every value in `AddRouteAnnotationRector`.